src/main/java/com/hms_networks/americas/sc/extensions/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -401,13 +387,67 @@ public static synchronized ArrayList getFifoNextSpanData(
     }
 
     // Store end time +1 ms (to prevent duplicate data)
-    final String newTimeTrackerVal = Long.toString(endTimeTrackerMsLong + 1);
-    FileAccessManager.writeStringToFile(writeFile, newTimeTrackerVal);
-
-    isFile1CurrTimeTrackerFile = !isFile1CurrTimeTrackerFile;
+    UpdateTrackingStartTime(endTimeTrackerMsLong + 1);
 
     // Return data
     return queueData;
   }
 
+  /**
+   * Get the Historical FIFO starting time from the tracking files
+   *
+   * @param startNewTimeTracker boolean to start a new time tracker
+   * @return tracking start time
+   * @throws IOException if unable to read or write files
+   * @throws TimeTrackerUnrecoverableException if both time tracking files are corrupted
+   * @throws CorruptedTimeTrackerException one of the tracking files is corrupted
+   */
+  private static long getTrackingStartTime(boolean startNewTimeTracker)
+      throws CorruptedTimeTrackerException, IOException, TimeTrackerUnrecoverableException {
+
+    /*
+     * The two time tracking files will swap on every iteration of grabbing new
+     * data points. Set the correct file using the isFile1CurrTimeTrackerFile flag.
+     */
+    String readFile;
+    String writeFile;
+    if (isFile1CurrTimeTrackerFile) {
+      readFile = timeMarkerFile1Name;
+      writeFile = timeMarkerFile2Name;
+    } else {
+      readFile = timeMarkerFile2Name;
+      writeFile = timeMarkerFile1Name;
+    }
+
+    // Get start time from file, or start new time tracker if startNewTimeTracker is true.
+    return getStartTime(startNewTimeTracker, readFile, writeFile);
+  }
+
+  /**
+   * Update Historical FIFO tracking time
+   *
+   * @param updateTime the time to be written
+   * @throws IOException if unable to read or write files
+   * @throws TimeTrackerUnrecoverableException if both time tracking files are corrupted
+   * @throws CorruptedTimeTrackerException one of the tracking files is corrupted
+   */
+  private static void UpdateTrackingStartTime(long updateTime)
```

Comment:
Method name does not follow the required format. It should be camel case, so

updateTrackingStartTime()

Reply:
updated.